### PR TITLE
Allow encryption of Kubernetes resources

### PR DESCRIPTION
**Pull request by bernardjkim**
## Description
These changes allow Kubernetes resources to be encrypted at rest. 

[aws-encryption-provider](https://github.com/kubernetes-sigs/aws-encryption-provider) binary is now built into Planet. A systemd unit file and an encryption-configuration file are also included. The configuration file can be found under `/etc/kubernetes/encryption-configuration.yaml`.

Planet now takes flags for:
- `--encryption-provider`
- `--aws-account-id`
- `--aws-key-id`
- `--aws-key-region`

If encryption is enabled, the `aws-encryption-provider` service is enabled and the `kube-apiserver` will be configured with the flag `--encryption-provider-config=/etc/kubernetes/encryption-configuration.yaml`

### CODE REVIEW

**Comment by wadells** [comment]
> Just some clean up. It looks like master-docker.mk is unused and is the same as planet.mk

Would you split these out into a separate PR please?  I'm all for clean up, but it gives a much cleaner history if orthogonal changes are independent.

**Review by wadells — CHANGES_REQUESTED** [review]
My big concern is the encryption provider build/version doesn't seem to be pinned.

Other concerns are minor.

File: Dockerfile [hunk 1]
```diff
@@ -349,6 +359,13 @@ RUN --mount=target=/host \
 	install -m 0755 /host/build.assets/makefiles/base/network/wait-for-flannel.sh /usr/bin/scripts && \
 	install -m 0755 /host/build.assets/makefiles/base/network/setup-etc.sh /usr/bin/scripts
 
+# encryption.mk
+COPY --from=aws-encryption-builder /aws-encryption-provider /usr/bin/aws-encryption-provider
+COPY ./build.assets/makefiles/encryption/aws-encryption-provider.service /lib/systemd/system/
+RUN set -ex mkdir -p /etc/kmsplugin
+RUN chmod o+t /etc/kmsplugin
+RUN chmod a+rwx /etc/kmsplugin
```
[the remark this answers is not in the document]
Comment (bernardjkim) [reply Dockerfile:367]: The `kube-apiserver` is running as `user=planet` and `group=planet`. So we needed a place to create the `encryption-provider.sock` socket file with planet ownership to allow the apiserver to connect to the socket. I'm not sure if this was the best way to handle this. If there is a better way, let me know.
Updated in 4600a60

File: build.assets/makefiles/encryption/aws-encryption-provider.service [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=AWS Encryption Provider
+Documentation=https://github.com/kubernetes-sigs/aws-encryption-provider
+
+[Service]
+EnvironmentFile=/etc/container-environment
+ExecStartPre=-/bin/rm /etc/kmsplugin/encryption-provider.sock
+ExecStart=/usr/bin/aws-encryption-provider \
+	--key=arn:aws:kms:${AWS_KEY_REGION}:${AWS_ACCOUNT_ID}:key/${AWS_KEY_ID} \
```
[the remark this answers is not in the document]
Comment (bernardjkim) [reply build.assets/makefiles/encryption/aws-encryption-provider.service:9]: The `kube-apiserver` is running as `user=planet` and `group=planet`. So we needed a directory to create the `encryption-provider.sock` socket file with planet ownership to allow the apiserver to connect to the socket. I'm not sure if this was the best way to handle this. If there is a better way, let me know.
Updated in 4600a60

File: build.assets/makefiles/encryption/encryption.mk [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+	cp -af $(BIN) $(ROOTFS)/usr/bin/aws-encryption-provider
+	cp -af ./aws-encryption-provider.service $(ROOTFS)/lib/systemd/system
+	mkdir -p $(ROOTFS)/etc/kmsplugin
+	chmod o+t $(ROOTFS)/etc/kmsplugin
+	chmod a+rwx $(ROOTFS)/etc/kmsplugin
+
+$(BIN):
+	mkdir /tmp/kubernetes-sigs
+	cd /tmp/kubernetes-sigs && git clone https://github.com/kubernetes-sigs/aws-encryption-provider --depth 1
```
[the remark this answers is not in the document]
Comment (bernardjkim) [reply build.assets/makefiles/encryption/encryption.mk:17]: The repo doesn't provide any tags. In this case do we want to just fork and create our own tag?

File: build.assets/makefiles/encryption/aws-encryption-provider.service (hunk 2 above)
[the remark this answers is not in the document]
Comment (wadells) [reply build.assets/makefiles/encryption/aws-encryption-provider.service:9]: Ok. I'm happy with current strategy.  It sounds like using only an ARN would probably introduce too much magic into the planet/gravity code, even if it does keep configuration simpler.

File: build.assets/makefiles/encryption/encryption.mk (hunk 3 above)
[the remark this answers is not in the document]
Comment (wadells) [reply build.assets/makefiles/encryption/encryption.mk:17]: No fork needed, but in lieu of tags, please check out a specific commit sha.

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (wadells) [reply Dockerfile:367]: This seems fine to me -- the comment was helpful.

File: build.assets/makefiles/encryption/encryption.mk (hunk 3 above)
[the remark this answers is not in the document]
Comment (bernardjkim) [reply build.assets/makefiles/encryption/encryption.mk]: Updated in 6df5190

**Review by wadells — APPROVED** [review]

**Pull request merged**